src/System.Management.Automation/engine/NativeCommandParameterBinder.cs

```diff
@@ -218,13 +218,117 @@ private void appendOneNativeArgument(ExecutionContext context, object obj, char
                         }
                         else
                         {
+#if UNIX
+                            // On UNIX systems, we expand arguments containing wildcard expressions against
+                            // the file system just like bash, etc.
```

Comment:
To be compatible as compatible with `bash` as reasonably possible. There are things `glob` does that we don't with character classes.

I don't recall discussing or looking closely at the inverse - if our wildcards support things that `glob` does not.